From afe7240976c988e7642b35c04e53224099b4479d Mon Sep 17 00:00:00 2001
From: Walter Neto <walter@tigera.io>
Date: Mon, 23 Sep 2024 20:26:23 -0300
Subject: [PATCH] Introduces TIGERA_TPROXY_ENABLED envvar

Now the TPROXY patch will only to have effect if this envvar exists
---
 source/extensions/filters/listener/original_dst/original_dst.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/extensions/filters/listener/original_dst/original_dst.cc b/source/extensions/filters/listener/original_dst/original_dst.cc
index 3bc718b28b..09cedd9916 100644
--- a/source/extensions/filters/listener/original_dst/original_dst.cc
+++ b/source/extensions/filters/listener/original_dst/original_dst.cc
@@ -26,7 +26,7 @@ Network::FilterStatus OriginalDstFilter::onAccept(Network::ListenerFilterCallbac
   case Network::Address::Type::Ip: {
     Network::Address::InstanceConstSharedPtr original_local_address;

-    if (Network::Utility::isTransparent(socket)) {
+    if (std::getenv("TIGERA_TPROXY_ENABLED") != nullptr && Network::Utility::isTransparent(socket)) {
       original_local_address = socket.connectionInfoProvider().localAddress();
     } else {
       original_local_address = getOriginalDst(socket);
-- 
2.40.1

